src/pools/selection-strategies.ts

```diff
@@ -106,25 +110,41 @@ class LessRecentlyUsedWorkerChoiceStrategy<
 }
 
 /**
- * Get the worker choice strategy instance.
+ * Selects randomly a worker once the pool is busy.
  *
- * @param pool The pool instance.
- * @param workerChoiceStrategy The worker choice strategy.
- * @returns The worker choice strategy instance.
+ * @template Worker Type of worker which manages the strategy.
+ * @template Data Type of data sent to the worker. This can only be serializable data.
+ * @template Response Type of response of execution. This can only be serializable data.
  */
-function getWorkerChoiceStrategy<Worker extends IWorker, Data, Response> (
-  pool: IPoolInternal<Worker, Data, Response>,
-  workerChoiceStrategy: WorkerChoiceStrategy = WorkerChoiceStrategies.ROUND_ROBIN
-): IWorkerChoiceStrategy<Worker> {
-  switch (workerChoiceStrategy) {
-    case WorkerChoiceStrategies.ROUND_ROBIN:
-      return new RoundRobinWorkerChoiceStrategy(pool)
-    case WorkerChoiceStrategies.LESS_RECENTLY_USED:
-      return new LessRecentlyUsedWorkerChoiceStrategy(pool)
-    default:
-      throw new Error(
-        `Worker choice strategy '${workerChoiceStrategy}' not found`
+class RandomWorkerChoiceStrategy<Worker extends IWorker, Data, Response>
+  implements IWorkerChoiceStrategy<Worker> {
+  /**
+   * Constructs a worker choice strategy that selects randomly once the pool is busy.
+   *
+   * @param pool The pool instance.
+   */
+  public constructor (
+    private readonly pool: IPoolInternal<Worker, Data, Response>
+  ) {}
+
+  /** @inheritdoc */
+  public choose (): Worker {
+    if (!this.pool.isDynamic()) {
+      const freeWorker = SelectionStrategiesUtils.findFreeWorkerBasedOnTasks(
+        this.pool
       )
+      if (freeWorker) {
+        return freeWorker
+      }
+    }
+    return this.pool.workers[this.getRandomInt(0, this.pool.workers.length - 1)]
+  }
+
+  private getRandomInt (max: number, min = 0): number {
```

Comment:
Should we use this implementation?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_integer_between_two_values

Reply:
The implementation in the PR is more optimized: one floor, one random only.

Reply:
But the need of an if else branch :thinking:  
An internal benchmark could be nice here again

Reply:
Ah you removed the branch